lib/LaTeXML/resources/XSLT/LaTeXML-webpage-xhtml.xsl

```diff
@@ -403,7 +403,13 @@
   <xsl:template match="/" mode="body-main">
     <xsl:text>&#x0A;</xsl:text>
     <xsl:element name="div" namespace="{$html_ns}">
-      <xsl:attribute name="class">ltx_page_main</xsl:attribute>
+      <xsl:attribute name="class">
+          <xsl:text>ltx_page_main</xsl:text>
+          <xsl:if test="//ltx:note[@role='margin'][@class='ltx_marginpar_reverse']"
+              > ltx_has_marginpar_reverse</xsl:if>
```

Comment:
Isn't there a `has()` CSS directive now that could be used?

Maybe @dginev has some comments on the CSS.

Reply:
[:has](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) is still missing from Firefox, so the recommendation I've heard is to use it for "progressive enhancement" only until it arrives. There is a single experiment in ar5iv with it, but I am mostly waiting for Firefox to ship it...

Reply:
Well, maybe we can avoid this `ltx_has_marginpar` complication with the assumption that the author will setup page dimensions to accommodate margins, whether or not there are marginpars? (you do in ar5iv).

Reply:
I'd like to have some styling so that there would be a possibility that a user could use the default style file.  It does appear that [Firefox will implement :has soon](https://connect.mozilla.org/t5/ideas/when-is-has-css-selector-going-to-be-fully-implemented-in/idi-p/23794), although that thread starts with "first half of 2023".  Maybe go with :has anyway?  That should be fully functional soon enough, and even if not, it would show the user what their style file could have.